sdk/python/tests/compiler/compiler_tests.py

```diff
@@ -124,6 +124,14 @@ def test_timeout_workflow(self):
     from .testdata.timeout import timeout_sample_pipeline
     self._test_pipeline_workflow(timeout_sample_pipeline, 'timeout.yaml')
 
+  def test_resourceOp_workflow(self):
+    """
+    Test compiling a resourceOp basic workflow.
+    """
+    from .testdata.resourceop_basic import resourceop_basic
+    nf = lambda f: re.sub("{},{.metadata.name}", "{.metadata.name},{}", f)
```

Comment:
Can we document why we need to do this substitution? Also, how can a user know what to define for a normalize function for resourceOp outputs? Also is it possible to have one generic regular expression for doing all these substitution?

Reply:
Add to the document. Thanks @Tomcli